### ISSUE TITLE
Many marine Siphonophora occurrences matched to milipedes

### ISSUE THREAD

**Issue by gbif-portal** [gbif/portal-feedback#4361]
**Many marine Siphonophora occurrences matched to milipedes**

Many marine Siphonophora occurrences are only given as genus Siphonophora which is a homonym and the accepted name are milipedes. From 11142 records over 10k are marine and therefore not milipedes.

-----

User: [See in registry](https://www.gbif.org/api/feedback/user/77afe531c4eaabb34a72162fb6e41753:006a54cd871258e41e84fa433cdca1f63638f9b3e9d4c1888872f03b106e09ef02c533ce6fe2b17c75b3eec6bfae33a1cf04aeae40334c5e9f39d44eac85e381aa9e25bfd8f0e818e565206c9e08b6c257fe63bf5714d1e078df1f76d4e00779fa592e64b75eba2189368bf0cf7948a7e3b6e1f9ad9849e66d54af420ee12906) - [Send email](https://www.gbif.org/api/feedback/user/mailto/77afe531c4eaabb34a72162fb6e41753:006a54cd871258e41e84fa433cdca1f63638f9b3e9d4c1888872f03b106e09ef02c533ce6fe2b17c75b3eec6bfae33a1cf04aeae40334c5e9f39d44eac85e381aa9e25bfd8f0e818e565206c9e08b6c257fe63bf5714d1e078df1f76d4e00779fa592e64b75eba2189368bf0cf7948a7e3b6e1f9ad9849e66d54af420ee12906)
System: Safari 15.4.0 / Mac OS X 10.15.7
Referer: https://www.gbif.org/occurrence/taxonomy?taxon_key=1011357
Window size: width 1276 - height 837
[API log](https://private-logs.gbif.org/app/discover#/?_g=(filters:!(),refreshInterval:(pause:!t,value:0),time:(from:'2022-10-24T09:48:10.437Z',to:'2022-10-24T09:54:10.437Z'))&_a=(columns:!(_source),filters:!(),index:'3390a910-fcda-11ea-a9ab-4375f2a9d11c',interval:auto,query:(language:kuery,query:''),sort:!()))
[Site log](https://private-logs.gbif.org/app/discover#/?_g=(filters:!(),refreshInterval:(pause:!t,value:0),time:(from:'2022-10-24T09:48:10.437Z',to:'2022-10-24T09:54:10.437Z'))&_a=(columns:!(_source),filters:!(),index:'5c73f360-fce3-11ea-a9ab-4375f2a9d11c',interval:auto,query:(language:kuery,query:''),sort:!()))
System health at time of feedback: OPERATIONAL

**Comment by mdoering** [comment]
See https://mobile.twitter.com/rdmpage/status/1584121572045905920 and this helpdesk mail 👍 

> GBIF Diplopoda went crazy: https://myriatrix.myspecies.info/myriatrix/diplopoda
> The problem occurs at the level of genus Siphonophora Brandt, 1837: https://myriatrix.myspecies.info/myriatrix/siphonophora
> Please, isolate any marine "Siphonophora" occurrences away from Diplopoda. Check for potential higher rank homonyms, e.g., "order Siphonophora". If that is the exact problem, prune the marine occurrences and reconnect them under order Siphonophorae (Animalia: Cnidaria: Hydrozoa): https://www.marinespecies.org/aphia.php?p=taxdetails&id=1371
> 
> I am not concerned that much about the effect in Diplopoda downloads, as the error will be noticed and cleaned. I am more worried about marine data and Cnidaria data downloads, where the error may go unnoticed. 
> The issue was reported through a Myriapodology mailing list and I would like to let them know as soon as it is fixed.

**Comment by mdoering** [comment]
> ALA and GBIF have independently (different taxonomic lookups) confused Siphonophora (millipede genus) with Siphonophora (colonial jellyfish class).

![image](https://user-images.githubusercontent.com/327505/197500684-a4dbf949-7427-4a9e-ae8f-2129a027a5cb.png)

![image](https://user-images.githubusercontent.com/327505/197500733-e6613a35-14ea-4b8e-ad14-afdd71b462b3.png)

**Comment by mdoering** [comment]
many marine datasets involved: 
<img width="401" alt="image" src="https://user-images.githubusercontent.com/327505/197502668-b925e8e0-b79a-48e6-be8f-d4ffb6ddedf7.png">

There are actually 3 Siphonophora genera in the Backbone:
https://www.gbif.org/species/search?q=Siphonophora&rank=GENUS&qField=SCIENTIFIC
<img width="886" alt="image" src="https://user-images.githubusercontent.com/327505/197502943-ffde8535-c197-4f2f-b2c6-43e26735a400.png">

The accepted Diplopoda, a Diptera synonym and a doubtful Hemiptera name.

An example of a [marine record](https://www.gbif.org/occurrence/3913945433) has nothing but scientificName= Siphonophora. No further classification to disambiguate. 

We can add Siphonophora as a synonym to the backbone (see [WoRMS](https://www.marinespecies.org/aphia.php?p=taxdetails&id=1371)) to potentially catch those records, but @ahahn-gbif it needs at least some indication which Siphonophora is meant. The marine ones should have rank=order at least to disambiguate them from the other genera!

**Comment by mdoering** [comment]
It would really be desirable to have the name matching be environment aware, i.e. you could supply an optional parameter marine, freshwater, terrestrial to disambiguate taxa of those environments only. COL has started to capture this, but it is not complete. It seems by just keeping sea vs land distinct we would have a very valuable additional hint for the matching

**Comment by mdoering** [comment]
https://www.gbif.org/species/203130334

**Comment by mdoering** [comment]
@ahahn-gbif @timrobertson100 The [plankton datasets](https://www.gbif.org/dataset/36449c1f-679d-4235-b34e-1c275ebcd968/metrics) involved here contain all kinds of taxonomic groups, so it is impossible to provide dataset wide configurations that set the classification and give matching hints:

<img width="1351" alt="image" src="https://user-images.githubusercontent.com/327505/197543319-ad40e3da-f5e3-48a6-96bc-c3487a616424.png">

Assuming the dataset publishers also do not know more we are pretty much stuck in this situation with current tooling.

What if we had negative dataset wide configurations for taxonomic coverages? If we could declare that this dataset never contains any Diplopoda? The name matching could then receive some new exclusion filter parameter that would allow to snap to the right Siphonophora. Such a config would likely help in a lot of cases when we receive bad matching reports and should not be terribly difficult to implement

**Comment by timrobertson100** [comment]
Makes sense to implement, although here I suggest we start by asking the publishers to provide some higher taxa if they can. 

With the forthcoming backbone, this will at least be an ambiguous lookup and snap to Animalia I presume, right?

**Comment by Archilegt** [comment]
See also how one of four Siphonophora treatments displays no data. 
https://www.gbif.org/species/1011357/treatments
Is that a Plazi treatment? How many broken treatments like that are there in GBIF?

![image](https://user-images.githubusercontent.com/60008120/197567174-60b484c2-4115-43fe-b886-59e07f742cc0.png)

**Comment by timrobertson100** [comment]
I'm in favor of 2 as well, and the machine tags are ideally suited to this.

I'm not sure we expect the matching service to have any knowledge of datasets, nor the performance hit of pulling dataset metadata. Perhaps the match should allow the caller to pass in an `exclusionScope` as a parameter, and the pipelines (which does know about the dataset it is processing) pull the tags and add it? (you may have meant that)

**Comment by mdoering** [comment]
Yes, that was exactly what I was thinking. Pipelines would consult the registry to find exclusions (or default classifications) that are then passed to the matching

**Comment by timrobertson100** [comment]
That seems like a reasonable approach to me.

One alternative I can think of is the ability to inject custom interpretation. Imagine a machine tag allowing the registration of a function (e.g. JS using [nashorn](https://docs.oracle.com/javase/10/nashorn/introduction.htm#JSNUG136)):

```
if (occurrence.scientificName === 'Siphonophora' && occurrence.phylum = null) {
  occurrence.phylum='Cnidaria'
}
```

This would open the door to selectively fixing data and is something we've pondered before but discounted as a step too far and could be dangerous.

**Comment by mdoering** [comment]
the v2 matching now includes a `exclude` parameter that takes (higher) nub keys. Any potential matches against those will be removed from results.

**Comment by mdoering** [comment]
There is a first test version running on the new backbone:

This yields millipeds as usual if nothing but the name `Siphonophora` is given:
http://backbonebuild-vh.gbif.org:9000/species/match2?verbose=true&name=Siphonophora

But you can exclude millipedes now:
http://backbonebuild-vh.gbif.org:9000/species/match2?verbose=true&name=Siphonophora&exclude=361

... which unfortunately only snaps to Animalia as there are many other Sinophora names in the nub still :(
 - order Siphonophora: synonym of Siphonophorae in Hydrozoa (desired match)
 - genus Siphonophora Meijere, 1909: Diptera synonym of Siphonophysa Hendel, 1907
 - genus Siphonophora Koch, 1855: Hemiptera
 - order Siphonophorae: Hydrozoa (fuzzy, but correct)
 - genus Siphonophera Senowbari-Daryan, 1982: Protozoa
